diff -Naur ncurses-6.3/config.sub ncurses-6.3-patched/config.sub
--- ncurses-6.3/config.sub	2021-08-14 17:20:38.000000000 +0100
+++ ncurses-6.3-patched/config.sub	2022-09-17 01:10:02.756226186 +0100
@@ -1746,7 +1746,7 @@
 	     | scout* | superux* | sysv* | rtmk* | tpf* | windiss* \
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
-	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
+	     | onefs* | tirtos* | phoenix* | fuchsia* | onyx* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr*)
 		;;
diff -Naur ncurses-6.3/configure ncurses-6.3-patched/configure
--- ncurses-6.3/configure	2021-10-17 16:12:23.000000000 +0100
+++ ncurses-6.3-patched/configure	2022-09-17 01:10:02.769559685 +0100
@@ -3462,8 +3462,6 @@
 echo $ECHO_N "checking if $CXX works... $ECHO_C" >&6
 
 		save_CPPFLAGS="$CPPFLAGS"
-		eval cf_includedir=${includedir}
-		CPPFLAGS="$CPPFLAGS -I${cf_includedir}"
 
 cat >"conftest.$ac_ext" <<_ACEOF
 #line 3469 "configure"
@@ -6299,7 +6297,7 @@
 		fi
 		cf_cv_rm_so_locs=yes
 		;;
-	(linux*|gnu*|k*bsd*-gnu)
+	(linux*|gnu*|k*bsd*-gnu|onyx*)
 		if test "$DFT_LWR_MODEL" = "shared" && test -n "$LD_RPATH_OPT" ; then
 			LOCAL_LDFLAGS="${LD_RPATH_OPT}\$(LOCAL_LIBDIR)"
 			LOCAL_LDFLAGS2="$LOCAL_LDFLAGS"
