diff -Naur bim-3.0.0/Makefile bim-3.0.0-patched/Makefile
--- bim-3.0.0/Makefile	2021-11-18 21:11:46.000000000 +0000
+++ bim-3.0.0-patched/Makefile	2022-04-26 19:15:00.077999942 +0100
@@ -1,5 +1,5 @@
 TARGET=bim
-CFLAGS=-g -O2 -std=c99 -Wvla -pedantic -Wall -Wextra -I. $(shell docs/git-tag) -Wno-unused-parameter -Wno-unused-result
+CFLAGS+=-g -O2 -std=c99 -Wvla -pedantic -Wall -Wextra -I. $(shell docs/git-tag) -Wno-unused-parameter -Wno-unused-result
 LDFLAGS=-rdynamic
 
 ifeq (Darwin,$(shell uname -s))
@@ -29,6 +29,7 @@
 *.o: $(HEADERS)
 
 bim: bim.o
+	$(CC) $(CFLAGS) $(LDFLAGS) $(LDLIBS) $< -o $@
 
 clean:
 	-rm -f $(TARGET) bim.o $(SYNTAXES)
